### PR TITLE
Issue/2303 - Fix pagination does not work if Affiliate Area is set to be the "front page"

### PR DESCRIPTION
**Pull request by tubiz**
Issue: #2303

### CODE REVIEW

**Comment by tubiz** [comment]
@amdrew PR updated.

**Comment by pippinsplugins** [comment]
I approve of `affwp_get_current_page_number()`. Definitely helps simplify it.

**Comment by tubiz** [comment]
I will add the test for the new function.

**Comment by amdrew** [comment]
PR update looks good, can be merged after test added.

**Comment by tubiz** [comment]
@DrewAPicture I am stuck in writing the test for the `affwp_get_current_page_number()`. Can you direct me?

**Comment by DrewAPicture** [comment]
@tubiz Let me take a look.

File: includes/misc-functions.php [hunk 1]
```diff
@@ -1233,3 +1233,20 @@ function affwp_maybe_unserialize( $original ) {
...
+/**
+ * Retrieves the current page number.
+ *
+ * @since 2.1.12
+ *
+ * @return int The current page number.
+ */
+function affwp_get_current_page_number() {
+	if ( is_front_page() ) {
+		$page = get_query_var( 'page' ) ? get_query_var( 'page' ) : 1;
```
Comment (DrewAPicture) [inline includes/misc-functions.php]: `get_query_var()` accepts a second optional `$default` parameter, so you could simplify these two ternaries.

**Comment by DrewAPicture** [comment]
@tubiz @amdrew Yeah, so this is actually super hard to test and not really possible to make it consistent. So we should probably just skip the test(s) since we're really just working around a quirk of WP core anyway.

**Comment by amdrew** [comment]
@DrewAPicture 👍

File: includes/misc-functions.php (hunk 1 above)
Comment (tubiz) [reply includes/misc-functions.php]: I will update the PR shortly.

**Comment by tubiz** [comment]
@amdrew @DrewAPicture PR updated. Thanks.

**Pull request merged**